dapr-spring/dapr-spring-boot-autoconfigure/src/main/java/io/dapr/spring/boot/autoconfigure/pubsub/DaprPubSubAutoConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.spring.boot.autoconfigure.pubsub;
+
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.spring.boot.autoconfigure.client.DaprClientAutoConfiguration;
+import io.dapr.spring.messaging.DaprMessagingTemplate;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+@AutoConfiguration(after = DaprClientAutoConfiguration.class)
+@ConditionalOnClass(DaprClient.class)
+@EnableConfigurationProperties(DaprPubSubProperties.class)
+public class DaprPubSubAutoConfiguration {
+
+  @Bean
```

Comment:
@artur-ciocanu @ThomasVitale the more I look into this.. I believe we don't need it.. this is @Configuration that the user should provide. -> https://docs.spring.io/spring-kafka/reference/kafka/sending-messages.html To quote Spring Kafka docs: 
```
To use the template, you can configure a producer factory and provide it in the template’s constructor. The following example shows how to do so:

@Bean
public ProducerFactory<Integer, String> producerFactory() {
    return new DefaultKafkaProducerFactory<>(producerConfigs());
}

@Bean
public Map<String, Object> producerConfigs() {
    Map<String, Object> props = new HashMap<>();
    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
    // See https://kafka.apache.org/documentation/#producerconfigs for more properties
    return props;
}

@Bean
public KafkaTemplate<Integer, String> kafkaTemplate() {
    return new KafkaTemplate<Integer, String>(producerFactory());
}
```

Reply:
@artur-ciocanu My recommendation will be to remove the PubSub and KeyvalueTemplate autoconfigurations, and have tests that shows that if the user adds the right beans when they want to use the messagingTemplate and the keyvalueTemplate @beans everything works. 
For @EnableDaprRepositories is different because we are explicitly enabling a feature that comes with default keyvaluetemplate instances.

Reply:
I have moved the `org.springframework.boot.autoconfigure.AutoConfiguration.imports` to starter and this should resolve the issue. The starter is not included in the `sdk-tests`.